### PR TITLE
Include auction assets

### PR DESCRIPTION
**Pull request by mootz12**
* allow auction creates the ability to specify bid and lot assets in an auction
* add protections against potential misuse of assets to create weird / incomplete liquidations

### CODE REVIEW

File: pool/src/auctions/bad_debt_auction.rs [hunk 1]
```diff
@@ -8,12 +8,25 @@ use crate::{
...
-    if storage::has_auction(e, &(AuctionType::BadDebtAuction as u32), backstop) {
+pub fn create_bad_debt_auction_data(
+    e: &Env,
+    user: &Address,
+    bid: &Vec<Address>,
+    lot: &Vec<Address>,
+    percent: u32,
+) -> AuctionData {
+    let backstop = storage::get_backstop(e);
+    if user.clone() != backstop {
```
Comment (Ryang-21) [inline pool/src/auctions/bad_debt_auction.rs:23]: ditto

File: pool/src/auctions/user_liquidation_auction.rs [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
 pub fn create_user_liq_auction_data(
     e: &Env,
     user: &Address,
-    percent_liquidated: u64,
+    bid: &Vec<Address>,
+    lot: &Vec<Address>,
+    percent: u32,
 ) -> AuctionData {
+    let user_clone = user.clone();
+    if user_clone == e.current_contract_address() || user_clone == storage::get_backstop(e) {
```
Comment (Ryang-21) [inline pool/src/auctions/user_liquidation_auction.rs:21]: ditto

**Review by Ryang-21 — CHANGES_REQUESTED** [review]

File: pool/src/auctions/auction.rs [hunk 3]
```diff
@@ -38,80 +38,82 @@ impl AuctionType {
...
+            let auction_data = create_interest_auction_data(e, user, bid, lot, percent);
+            storage::set_auction(
+                e,
+                &(AuctionType::InterestAuction as u32),
+                &user,
+                &auction_data,
+            );
+            auction_data
+        }
     }
```
[the remark this answers is not in the document]
Comment (mootz12) [reply pool/src/auctions/auction.rs:116]: `AuctionType::from_u32(e, auction_type)`

will panic if it is not a valid auction type, so no default case applies, but I should probably ensure that it unit tested appropriately

File: pool/src/auctions/backstop_interest_auction.rs [hunk 4]
```diff
@@ -10,46 +10,64 @@ use super::{AuctionData, AuctionType};
 
 pub fn create_interest_auction_data(
     e: &Env,
-    backstop: &Address,
-    assets: &Vec<Address>,
+    user: &Address,
+    bid: &Vec<Address>,
```
[the remark this answers is not in the document]
Comment (mootz12) [reply pool/src/auctions/backstop_interest_auction.rs:14]: it's required for user liquidation auctions, so for consistency it's used for creating all auctions

**Review by markuspluna — APPROVED** [review]
lgtm

**Review by Ryang-21 — APPROVED** [review]

**Pull request merged**